tests/components/totalconnect/test_config_flow.py

```diff
@@ -0,0 +1,108 @@
+"""Tests for the iCloud config flow."""
+from unittest.mock import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components.totalconnect.const import (
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+
+from tests.common import MockConfigEntry, mock_coro
+
+USERNAME = "username@me.com"
+USERNAME_2 = "second_username@icloud.com"
+PASSWORD = "password"
+
+
+async def test_user(hass):
+    """Test user config."""
+    # no data provided so show the form
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=None
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    # now data is provided, so check if login is correct and create the entry
+    with patch(
+        "homeassistant.components.totalconnect.config_flow.TotalConnectConfigFlow.is_valid",
+        return_value=mock_coro(True),
+    ):
```

Comment:
Please patch the library code instead of our code in the config flow. We want to test the config flow code.
```suggestion
        "homeassistant.components.totalconnect.config_flow.TotalConnectClient.TotalConnectClient"
    ) as client_mock:
    client_mock.return_value.is_logged_in.return_value = True
```

Reply:
good point, thanks